cdap-runtime-spi/src/main/java/io/cdap/cdap/runtime/spi/provisioner/Provisioner.java

```diff
@@ -73,17 +73,17 @@ default void initialize(ProvisionerSystemContext systemContext) {
   Cluster createCluster(ProvisionerContext context) throws Exception;
 
   /**
-   * Get the status of the cluster. If it does not exist, {@link ClusterStatus#NOT_EXISTS} should be returned.
+   * Get the status of the cluster. If it does not exist, status should be set to {@link ClusterStatus#NOT_EXISTS}.
    * This is used to poll for status after {@link #createCluster(ProvisionerContext)} or
    * {@link #deleteCluster(ProvisionerContext, Cluster)} is called.
    *
    * @param context provisioner context
    * @param cluster the cluster to get the status of
-   * @return the status of the cluster
+   * @return cluster with status set.
    * @throws RetryableProvisionException if the operation failed, but may succeed on a retry
    * @throws Exception if the operation failed in a non-retryable fashion
    */
-  ClusterStatus getClusterStatus(ProvisionerContext context, Cluster cluster) throws Exception;
+  Cluster getClusterStatus(ProvisionerContext context, Cluster cluster) throws Exception;
```

Comment:
we can't change methods in any api or spi module. This is backwards incompatible and will break all existing provisioners.

Reply:
No changes in the existing API now.